File: frontend/src/services/search.service.ts
import api from './api';
import type { Page, SearchResult } from '@/types/api';

class SearchService {
    search(query: string, projectId: string | null, page = 0, size = 20) {
        const types = "point, polygon, multiline";
        return api.get<Page<SearchResult>>('/search/geo', { params: { query, types, page, size, projectId } });
    }
}

export default new SearchService();

File: search-service/src/main/java/kg/geoinfo/system/searchservice/service/SearchService.java
package kg.geoinfo.system.searchservice.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Map;

public interface SearchService {

    Page<Map> search(String query, Pageable pageable);

    Page<Map> searchGeoObjects(String query, java.util.List<String> types, Pageable pageable, String projectId);

}